src/adservice/build.gradle

```diff
@@ -14,14 +14,15 @@ description = 'Ad Service'
 group = "adservice"
 version = "0.1.0-SNAPSHOT"
 
-def opentelemetryVersion = "1.12.0"
+def opentelemetryVersion = "1.19.0"
+def opentelemetryInstrumentationAlphaVersion = "1.19.1-alpha"
 def grpcVersion = "1.45.1"
 def jacksonVersion = "2.13.2"
 def protocVersion = "3.20.0"
 
 tasks.withType(JavaCompile) {
-    sourceCompatibility = JavaVersion.VERSION_18
-    targetCompatibility = JavaVersion.VERSION_18
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
```

Comment:
I agree that it's good for demo to use latest Java LTS👍

Reply:
(one could even argue for it being Java 11 based on https://newrelic.com/resources/report/2022-state-of-java-ecosystem#toc-java-11-is-the-new-standard and the low Java 17 adoption rate, but I personally agree with using Java 17)

Reply:
I think choosing the latest LTS is simple and defensively. Latest non-LTS version will be prone to issues with tooling, and choosing a non-latest LTS (i.e. 11) would seem kind of arbitrary.